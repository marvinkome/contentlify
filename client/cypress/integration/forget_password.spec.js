describe('Forget Password Page', () => {
    beforeEach(() => {
        cy.visit('/auth/forgot-password');
    });

    it('should display forgot password form', () => {
        cy.get('.auth__container').toMatchSnapshot();
    });

    it('should submit form to api', () => {
        // server setup
        cy.server();
        cy.route({
            method: 'POST',
            url: '/api/auth/forget-password',
            response: { message: 'Test' }
        }).as('forgetPasswordApi');

        // type in email
        cy.get('input#email').type('testuser@gmail.com');

        // click button
        cy.get('[data-testid="submit-btn"]').click();

        // check that request has been made to the server
        cy.wait('@forgetPasswordApi').then((xhr) => {
            // check that request was made
            assert.deepEqual(xhr.request.body, { email: 'testuser@gmail.com' });
            assert.isNotNull(xhr.response.body, 'Forgot password API Called');
        });
    });
});
