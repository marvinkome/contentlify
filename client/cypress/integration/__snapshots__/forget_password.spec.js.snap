exports[`Forget Password Page > should display forgot password form #0`] = `
<main class="auth__container">
  <section class="form-section">
    <header class="header">
      <a href="/"
        ><img
          alt="logo"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAYIAAABmCAYAAADGWJapAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAA7uSURBVHgB7d3tdds4Fgbgd+bs/8lWIEwF41QQpIKxKzBdQbIVGKnAdgVSKrBTAZUKnK2A3AqcVKDVFaVY40gWLnBBSdT7nIPjOGEAiCIBEp8AEREREREREREREREREREREREREREREREREREREREREREREREREREREREREREN0G8b/u58Hv6eB4fT8n0evszDBEREJ+psHpp5mJ14aHB6lSAR0aISeMJxFdglw9PynBARnYQ34JvAptAszw0R0aD9jq5PwIFecvPwEUREAycVwSVom3cgIho4GTU0A20jI4n+DSKiAdt7ReCcw9nZGd686Zrj5ef3798Xf27bdvHnb9++YY9+AxHRwPXaCeu9n93c3Mzqup7Fenp6Whz/8ePH2bzi6LvTmIho0Hp5I5Cn/A8fPmBekP988s8xnU7x+fNnTCYT9KCvNwKHbsiq/JST9Mfy738sf8prkbwqTUFEQ+TmwaO7/1+WAXLvt8tg3kRSvCKQCiCEYFIBvCRNRxcXF6WbjkpVBHJCZMTWu+VPzQmaopsF/YDuwrB0j/g5FHfzcIt8MjrrQ+Sx29JscBim83C14e8157VUHoSbhxpxpPC5gM01ViN+dOKfG/5Oc42UJufEotCRe75Ct5LDGeLLgO/L9D+j+65bGCjSpCJNOI+Pj7M+SFPTvKI5lqYhNw83sJvAV6O7mKzUyvQtCregSC9siWN2IKHekr/6APIgnGFcGo0izU0CdPkuGTzyeNheD2MYTAEwP1GXl5eLdv0+NU1Tqv/AitT2N7DP3yo0sJkPUivTfUS+oEgvbIljdiCh3pK/+gDyIFxCfBbzaRpFepsE6PNdKnikcSh7HYyRWAb8DmPX19eLtvsSTUGvkdFH8w7lxc8DJE0/DcpOUHPLNKSy6fPkyxtBAA3ZNTjpNJc0a8lDk0c5Fboy4BoJzGqkeSUw27cCbwa5Sr4FbAsN0m/cOiG9J+QVFEGRVtgSx+xAQr0lf/UB5EG4AnHGaBRpbRKgz3Op4BFPHsrqPeTxHooHQrM3gvPz80Wn8L7JG8H9/X3vbyQbSAbG2M8yFQ7dxddX5+Tqs9Jwedj2RZ0Ch/JvAducL9N2MQebVARS+M47bHEoZIKaNFHtmRSMFfbHQTdSI5cH12Yaur6bHY+ZQ7/3X1Ye/gUDUujmtM3L8E8JP378WMwkHo1Gi/i890glcxa+fPmymHOwB3LDnGP/Vq+l72E/zHQTqX1LDGmlw7B687sA7SJNMw7759DlRcqA79sOyq4IqqpaBC0p8O/u7nB7e/tzSYmXpHlHmpxSKxr5f3uoCKQwPKQnY4fu5n2P8t70mNZLLXSc4thWcex35GuRxyIP25wvwwP6tZpQFWs1KSs2bstzJg+CfTXLxpC8SLn0n9cOyuqUkM5ZrXkbvnrc/7z/YZZClrTI/IwaLjOtBt1kKalIqrUQkN/hFFs55aajSWslKOIOyOcU6TWwoTmvI5TjFPnYFmRwgLaJqFHEbyEo0qtgpwLU53M91Pi1DPi4/Ls6M27/WsaTI56/Ccy0pEDvMz1Zoyjz5Gk0iWmMEfcE4fA8REybRuzInhppnyElrZWgiDsgn1Ok18CG5ryOUI5T5GPXNavRKOK2EBTpVbDhkH5vBsRVrg7dvuop31nzWhrJF4Nm4TgxbwbKvgBT3gwyZx3HqhLibpA2osCha/fTpjeOiLtOiHdTqBEvKOINyOcU6TWwoTmvI5TjFPnYFTziNYp4LQRFehVsXAPqc1gjrS/BIa3SCdsiTLoIZKy+hjQhWS0Doa2AZNXSjPRiNcp45XiHPLfQfx63I846Ic5tIbaJKCjiDMjnFOk1sKE5ryOU4xT5iDk3sU1EjSJeC0GRXoV8DvoyYII8cu4flWlubNZL7izWjuj59OnT1k5hraurK1XnsSxOV5iHvvPRYiSPFLR/QfdkVqG/mcDX4CiiIXOI6IQ8ER66MuAb8isgKVClHImeL4DnxS4nL/8h6WlgPB5HP5HL20BqOgcQYoyVcVawI1+sZgG7px3x1dB9ll2hxm5BEV9APqdIr4ENzXkdoRynyEds8BHpNor4LARFehXyaZ/MHex4Zdr1ywiSJ5TJpK1YexrL3yevOHaK/FfCdfJUcKc4XioOj/54cKLZ0I1x2hPNHHTDRSewfUueoluSOpbHi++rl4pANpEZMA9d7a4ptGNJX4Gm3c2jX9fgomVD5pC40NlAeOh8gr1b5fF+/ZekikA7uWvPew6XpnkSkMK6xEQciVdT21pNdplGHrdagpviyMYsLjFYWu2KF0Pe+jxO0zvFsVOU6TP7Bt3uhX79l6TOYm1FYNVJfKCc4tiSNaIm7r9gQ95uHOLOwb5mpB6jGmlabN7dK5XcuPL0GluJSxPRW5Sd2XyInOLYryjnv4ivjN36L+b7EbzUw4idfdMUqiUvgqniWAebNl254TUjRk69LfkYSZPDNPJYh9NsItK8YR/kw2DxiuAEaAq2FuVon8L+gI0HxD/lc7nq4yT7H7OJaDtNGVDybUlTEdh0FtNPh/KEu8/XcU1BIc1DHnRMWug6OE/pzS9lzaVSNGVAfkWgafM/0K0jyb4t+UpxPJuIjg+biDYbxHVcvCIQsr8AHRzrJxNNE5HDaQ83PFZsIvrVIXWMJ1dKSRWBdABrKoO3b99iwJJfx4wdwpMJC4pha6FrIjqFIcPaisDyTfyl5L6K5D4CzWignJ3GjkCrONahHAed/8Eem4iGT9NEdIb+1rXap1ZxrEM5mtFL7fovyYvOff36NXp28eXl5WLryIHSFKhW4/c3OZQhbNI8NEXc075D10T0A7RObpaUc9JXM4VU9rK2TkwlLt9vbJPhsWoRX8Br7lMtTfnyj3IruSLQzBaWLSdlO8vJZAILUgFplriQt5eC6x1pClXJtNw8JW7YvxXHlngbWKcpKKTQm4LWScFZ+jvK0UI/0WzINBO55D4t9VTsFcf+Um7NUoLsLaDx9PRkth+BdntM2dksI71dnDK+EheBM8xDrYjHvxLPR2WeYkNAPqdIr4ENzXkdoRynyEe9Iy7NZ9IEC0GRXoU8FXSfz8Oez8lDch+BdBZrnrLlrWA8HiNXykb2hVc/baF7wtc8uce6hM4U5Wnakuk4afqDhkzb9KW9X0vEafNGIEE2hteaF+TJ6cn/1Xp4eOjj6eRWGaflW4FTpt3siK9WxOWxO2+avRJiQkA+p0hv1/mKpTmvI5TjFPmoI+ILivhig4WgSK9Cvhq6z+hhp1KmvbHiyvrStNtGivv7+8VWl7FpSJPSzc3NLIVUVpmfMYZXxqnd3H0baYNvlGnvqoRqRVweu1k3EQXkc4r0GtjQnNcRynGKfNSRcWo3ZdkVLARFehXyaa/zBjZlgIO+DDjfFFHWl5byViCkz0B2OXutQpAKQPYb1vYJrEgllfv5EK9Wxtsg70KQSiDlBnSw+xwecbTn5rUQkM8p0mtgQ3MORijHKfJRR8Z5pogzJlgIivQq5NPuFChBs8XkJg5pe6VvlP3FyRN+DinopdCWimEVHh8fZ7k0bx3Ivyh9Qtxy4VTQ89BfABLGEXHXivg84jjYNREF5HOK9BrY0JzXEcpxinzUiniDIt5dwUJQpFfBRoD+szZIaybySLunqm0RZn9x8uSe+tReSghhHxdlnZhGs/yCdg239NDvj7yehoPtZ/CIZ9VEFJDPKdJrYENzXkcoxynyUUPHqonIQlCkV8FGSlPtKowRdz/JMXViGs22SJPnEayTEURXV1eYP9XjEMgch3lFgD3QjJ9f59BdCBKm6EYircaRj/C8z3DOLNxPKLsM9i7SoS4jpjxoqFbX/6lazaxPKQirZZA4ZERPi3+WAQ7P85BSvd/2DyYVgZAhmlIZWAwRzSGTxy4uLrAnLbqNWnJOgoe9yTLsW2pFeWomyCMFiWbDICuSrjxwXON0TdHt3PcBaVYPfdZ2PgjOLIN07u6LNE8Z9QvkvqYG4zzkhAa6grdWxO2hl9tEFJDPKdJrYENzXnND/Uo+nFE82+Q0j6yChaBIr4K9GnnnwDJMdmXWfGOa29vbxRN531tUSnPQ+/fvD2VrzADdZvKltOheB/tagyYGJ5oN26p55NRJs0TJNb1iSR52zlsqskPZw8PDolAuPKP3p7u7u8VS1we2P3IF3ZK91lp0lUCLw6NZrpqOzxRd88gpk+tb7r99VgZTKB4Ei76WyDo/pUYUyZDTs7Oz0q9VuQLK5m9TqJHeDl8r0vFIl9pEFJDPKdJrYENzXi2+/22cUTy75DQRWQiK9CqUpV15wCLcQqmXjEmFkDIL+SWZiCbxGMwYjg0WHPLbTWOCjCvOXb6iVqTnkUeT1ioE5HOK9BrYSPmsqaF+JR/OKJ4YHmn5txAU6VUor0J/ZcA5Esz6DNKZK5WCTEKLfVOQ42SSmfw/qxVMFcFSQLmLoYbNlPVakaZHHgf9pJiAfE6RXgMbmvNqcS1s44ziiZXyNGwhKNKr0A+HruO2xHcu91FAxoi82T6DFOzSvCNP+BKksF/9ucAIoJRgzcHu6UC+fLnRPOzUivQ98mmbiALyOUV6DWxozmtuqF/JhzOKJ1ZKE5GFoEivQr8cugpBe162lQEBmUOyzeYRpJLJaJpNbgagxfO4fo/uNe4d4ncuatF1An1Ft4rgsXe6cqLZsOVMshqqFs+Vz/kyyO5imjLgC7r7XwrP7DLgN9itgjdEcpLfoj9yITh0tfsqtHjOi/yZo22Ihknu99XsYQkO3f2+uueLlgEBZV9TjzmMQUR0AlKWTz2FYLVnABHRUahwXIV0HyF3GCYR0dGRgo9vBul7BBARrZPWFilXw/J3B/vyarSMu1n+Xi1/DxuO3Wp9iQkZvSEdo7JGzkkN41lq0S0J8ScOY6VOIjpOUgFcoyucb9CNCjxoL4ePtuDTMBFRKqkExkic3bsvRRadIyI6UTK0U1YelZYFaV1pcQT2PqGMiGiAWjy3rvjlT6kkJrD1Y/lTJpj9gX/OO5qAiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIiIhOwf8BxSCzl6n8yCYAAAAASUVORK5CYII="
          class="logo"
      /></a>
      <p>New to Contento? <a href="/register">Register</a></p>
    </header>
    <article class="form-container">
      <header class="form-header">
        <h1>Forgot your password?</h1>
        <p>
          Enter your email and we'll send you instructions on how to reset it.
        </p>
      </header>
      <form class="form">
        <div class="form-group">
          <label for="email">Email address</label
          ><input class="form-input" id="email" type="email" required="" />
        </div>
        <button type="submit" data-testid="submit-btn" class="btn btn-primary">
          Send reset link
        </button>
      </form>
    </article>
  </section>
  <aside class="about-section">
    <img
      alt="svg1"
      src="/static/media/auth-svg1.a68e392c.svg"
      class="svg-icon icon1"
    />
    <section class="about-information">
      <h1>A modern approach to content management</h1>
      <p>Finally, the freedom to manage your contents without changing code.</p>
    </section>
    <img
      alt="svg2"
      src="/static/media/auth-svg2.0ca4be78.svg"
      class="svg-icon icon2"
    />
  </aside>
</main>
`;
